src/expandableCalendar/WeekCalendar/presenter.spec.js

```diff
@@ -0,0 +1,359 @@
+      const context = makeContext(twoDaysDifferentWeek, UPDATE_SOURCES.PAGE_SCROLL);
+      const prevContext = makeContext(twoSomeDays);
+      const {shouldComponentUpdate} = makeUUT();
+      const shouldUpdate = shouldComponentUpdate(context, prevContext, 2);
+
+      expect(shouldUpdate).toBe(true);
+    });
+  });
+
+  describe("Month's array of dates by weeks, getDatesArray test", () => {
+    it('Expect to get array of dates before, and after the subject date', () => {
+      const index = 0;
+      const presenter = makeUUT();
+      const props = makeProps();
+      expect(presenter.getDate(props, index)).toEqual('2021-01-01');
+
+      expect(presenter.getDatesArray(props)).toEqual([
+        '2020-12-13',
+        '2020-12-20',
+        '2021-01-01',
+        '2021-01-03',
+        '2021-01-10'
+      ]);
+    });
+
+    it('Expect to get array of dates before and after the subject date, by passing `current` and without context', () => {
+      const contextDate = '2021-01-31';
+      const context = makeContext({date: contextDate});
+      const current = undefined;
+      const firstDay = 0;
+      const index = 0;
+      const presenter = makeUUT();
+      const props = {current, context, firstDay};
+      expect(presenter.getDate(props, index)).toEqual(contextDate);
+
+      expect(presenter.getDatesArray(props)).toEqual([
+        '2021-01-17',
+        '2021-01-24',
+        '2021-01-31',
+        '2021-02-07',
+        '2021-02-14'
+      ]);
+    });
+
+    it('Expect to get a week prior to current by passing negative index', () => {
+      const context = makeContext();
+      const firstDay = 0;
+      const presenter = makeUUT();
+      const props = {current: '2021-01-31', context, firstDay};
+      expect(presenter.getDate(props, 0)).toEqual('2021-01-31');
+
+      expect(presenter.getDate(props, -1)).toEqual('2021-01-24');
+      expect(presenter.getDate(props, -2)).toEqual('2021-01-17');
+      expect(presenter.getDate(props, -3)).toEqual('2021-01-10');
+    });
+
+    it('Expect to get a week after current week by passing positive index', () => {
+      const presenter = makeUUT();
+      const props = makeProps('2021-01-31');
+      expect(presenter.getDate(props, 0)).toEqual('2021-01-31');
+
+      expect(presenter.getDate(props, 1)).toEqual('2021-02-07');
+      expect(presenter.getDate(props, 2)).toEqual('2021-02-14');
+      expect(presenter.getDate(props, 3)).toEqual('2021-02-21');
+    });
+
+    it('Expect to be able to get past year', () => {
+      const presenter = makeUUT();
+      const props = makeProps('2021-01-07');
+      expect(presenter.getDate(props, -1)).toEqual('2020-12-27');
+    });
+
+    it('Expect to be able to get next year', () => {
+      const props = makeProps('2021-12-30');
+      const presenter = makeUUT();
+      expect(presenter.getDate(props, 1)).toEqual('2022-01-02');
+    });
+  });
+
+  describe('getMarkedDates test', () => {
```

Comment:
getMarkedDates() is gone...

Reply:
Should I use the `markedDates` directly now?

Reply:
Yes

Reply:
Please note that the 'selected' mark is now in the 'sate' prop and not updated in 'markedDates', so you probably need to test the 'this.props.state' "selected" string.

Reply:
Not sure I understood. Can't see `selected` or `this.props.state` in master -> WeekCalendar

Reply:
Correct. You don't need to see it anymore since it's managed by the Calendar/Week components now (getState() method). It used to be managed by modifying 'markedDates' object but I changed the implementation to avoid creating a new 'markedDate' object thus invoking unnecessary renders.

Reply:
Got it, anyway I removed those tests before.
To verify this logic, I believe it would be testing `getState` code